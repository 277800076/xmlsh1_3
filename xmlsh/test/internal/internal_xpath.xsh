# test of xpath

D1=<[
document {
<root a1="value1">
  <foo>
    <bar a2="value2"/>
    text
   </foo>
</root>
}
]>

cd ../../samples/data

#xpath on stdin
echo Xpath on stdin
xpath //AUTHOR < books.xml

#xpath on standard file
echo Xpath on file
xpath -i books.xml //AUTHOR

#xpath with xml expression
echo Xpath on xml expression
xpath -i $D1 //bar

