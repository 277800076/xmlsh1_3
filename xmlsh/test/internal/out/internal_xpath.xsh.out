Xpath on stdin
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jane Austen</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Charlotte Brontë</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR>
Xpath on file
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jane Austen</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Charlotte Brontë</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Thomas Hardy</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR>
<AUTHOR xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">Jasper Fforde</AUTHOR>
Xpath on xml expression
<bar a2="value2"/>
